xml/value.go

```diff
@@ -0,0 +1,339 @@
+
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// Float encodes v as a XML number.
+// It will auto close the xml element tag.
+func (xv Value) Float(v float32) {
+	writeStartElement(xv.w, xv.startElement)
+	xv.float(float64(v), 32)
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// Double encodes v as a XML number.
+// It will auto close the xml element tag.
+func (xv Value) Double(v float64) {
+	writeStartElement(xv.w, xv.startElement)
+	xv.float(v, 64)
+	writeEndElement(xv.w, xv.endElement)
+}
+
+func (xv Value) float(v float64, bits int) {
+	*xv.scratch = encodeFloat(v, bits)
+	xv.w.Write(*xv.scratch)
+}
+
+// Boolean encodes v as a XML boolean.
+// It will auto close the xml element tag.
+func (xv Value) Boolean(v bool) {
+	writeStartElement(xv.w, xv.startElement)
+	*xv.scratch = strconv.AppendBool((*xv.scratch)[:0], v)
+	xv.w.Write(*xv.scratch)
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// Base64EncodeBytes writes v as a base64 value in XML string.
+// It will auto close the xml element tag.
+func (xv Value) Base64EncodeBytes(v []byte) {
+	writeStartElement(xv.w, xv.startElement)
+	encodeByteSlice(xv.w, (*xv.scratch)[:0], v)
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// BigInteger encodes v big.Int as XML value.
+// It will auto close the xml element tag.
+func (xv Value) BigInteger(v *big.Int) {
+	writeStartElement(xv.w, xv.startElement)
+	xv.w.Write([]byte(v.Text(10)))
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// BigDecimal encodes v big.Float as XML value.
+// It will auto close the xml element tag.
+func (xv Value) BigDecimal(v *big.Float) {
+	if i, accuracy := v.Int64(); accuracy == big.Exact {
+		xv.Long(i)
+		return
+	}
+
+	writeStartElement(xv.w, xv.startElement)
+	xv.w.Write([]byte(v.Text('e', -1)))
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// Null encodes a null element tag like <root></root>.
+// It will auto close the xml element tag.
+func (xv Value) Null() {
+	// write open tag for the parent object
+	writeStartElement(xv.w, xv.startElement)
+
+	// close tag
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// Write writes v directly to the xml document
+// if escapeXMLText is set to true, write will escape text.
+// It will auto close the xml element tag.
+func (xv Value) Write(v []byte, escapeXMLText bool) {
+	writeStartElement(xv.w, xv.startElement)
+
+	// escape and write xml text
+	if escapeXMLText {
+		escapeText(xv.w, v)
+	} else {
+		// write xml directly
+		xv.w.Write(v)
+	}
+
+	writeEndElement(xv.w, xv.endElement)
+}
+
+// NestedElement returns a nested element encoding.
+// It returns a point to element object and a close function that will
+// close the element tag.
+func (xv Value) NestedElement() (o *Object) {
+	// write open tag for the parent object
+	writeStartElement(xv.w, xv.startElement)
+
+	return newObject(xv.w, xv.scratch, xv.endElement)
+}
+
+// Array returns an array encoder. By default, the members of array are
+// wrapped with `<member>` element tag.
+func (xv Value) Array() *Array {
```

Comment:
Should this example include the wrapping tag because this is being called via Value?

```xml
<someList><member>...</member></someList>
```